### PR TITLE
[quant][fx] Add BatchNorm ops to backend_config_dict

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #75378
* #75377
* #75318
* #75317
* #75316
* #75315
* #75314
* __->__ #75260
* #75259
* #75258
* #75241

Summary:
This PR adds bn2d, bn3d and corresponding bn - relu to backend_config_dict

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35403587](https://our.internmc.facebook.com/intern/diff/D35403587)

### CODE REVIEW

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35403587).

File: torch/ao/quantization/fx/backend_config/native.py [hunk 1]
```diff
@@ -351,6 +351,42 @@ def _get_binary_op_configs():
...
 
+def _get_bn_configs():
+    """ Get configs related to batchnorm
+    """
+    bn_configs = []
+    bn_to_fused_bn = {
+        torch.nn.BatchNorm2d: nni.BNReLU2d,
+        torch.nn.BatchNorm3d: nni.BNReLU3d,
+    }
+    for bn in [torch.nn.BatchNorm2d, torch.nn.BatchNorm3d]:
```
Comment (andrewor14) [inline torch/ao/quantization/fx/backend_config/native.py:362]: Can you just iterate through `bn_to_fused_bn.keys()`?

File: torch/ao/quantization/fx/backend_config/native.py [hunk 2]
```diff
@@ -351,6 +351,42 @@ def _get_binary_op_configs():
...
+        #     "dtype_configs": default_op_quint8_dtype_config,
+        #     "fuser_method": reverse_sequential_wrapper2(bn_to_fused_bn[bn]),
+        # })
+        bn_configs.append({
+            "pattern": bn,
+            "observation_type": ObservationType.OUTPUT_USE_DIFFERENT_OBSERVER_AS_INPUT,
+            "dtype_configs": default_op_quint8_dtype_config,
+        })
+
+    for fused_bn in [nni.BNReLU2d, nni.BNReLU3d]:
```
Comment (andrewor14) [inline torch/ao/quantization/fx/backend_config/native.py:382]: Similarly, is this just `bn_to_fused_bn.values()`?

**Review by andrewor14 — APPROVED** [review]

File: torch/ao/quantization/fx/backend_config/native.py (hunk 1 above)
Comment (jerryzh168) [reply torch/ao/quantization/fx/backend_config/native.py:362]: will fix in a future diff, I'm making some further changes down the road

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35403587).
